crates/runtime/src/traphandlers.rs

```diff
@@ -338,6 +338,28 @@ impl<T: Copy> Drop for ResetCell<'_, T> {
     }
 }
 
+/// Captures the current backtrace of the calling thread, ensuring that no more
+/// than a fixed number of frames are captured.
+#[inline(never)] // try to get this removed from stack traces
+pub fn capture_backtrace() -> Backtrace {
+    const MAX_FRAMES: usize = 100;
```

Comment:
Could you make this configurable? There are cases where long backtraces are not a bug. For example rustc recurses a lot in some places.